lib/ansible/plugins/__init__.py

```diff
@@ -90,6 +90,21 @@ def get_options(self, hostvars=None):
             options[option] = self.get_option(option, hostvars=hostvars)
         return options
 
+    def get_ff_option(self, *options, hostvars=None, pc=None):
+        ''' convenience function for when plugins have not followed strict conventions (user/remote_user)
+            and/or relied on play context, which is not strictly up to date, especially in loops. '''
+        for option in options:
+            if self.has_option(option):
+                return self.get_option(option, hostvars)
+
+        # fallback to play context
+        if pc is not None:
+            for option in options:
+                if hasattr(pc, option):
+                    return getattr(pc, option)
+
+        raise KeyError("The %s plugin '%s' does not support any of these options: %s" % (self.plugin_type, self._load_name, ', '.join(options))
```

Comment:
Hopefully, this'll help to spot mismatched parentheses faster:
```suggestion
        raise KeyError(
            f"The {self.plugin_type !s} plugin '{self._load_name !s}' does "
            f"not support any of these options: {', '.join(options) !s}",
        )
```

Reply:
Although, I'd go for a `LookupError` since it's semantically closer to what this case represents:
```suggestion
        raise LookupError(
            f"The {self.plugin_type !s} plugin '{self._load_name !s}' does "
            f"not support any of these options: {', '.join(options) !s}",
        )
```

Reply:
trying to keep the error the same as it is now, to keep changes at a min at the caller.

Reply:
The paren is still unclosed, though.
```suggestion
        raise KeyError("The %s plugin '%s' does not support any of these options: %s" % (self.plugin_type, self._load_name, ', '.join(options)))
```